### PR TITLE
Miawong/sc 71608/UI feedback from installing schrodinger s

### PR DESCRIPTION
**Pull request by miaawong**
#### What this PR does / why we need it: This PR address UI feedback from Schrodinger's App. 
- Removed extra preflight icon that was being displayed.
- Improve content layout for sequence number for readability.
- Fixed view diff behavior to improve content layout, and added a back button to prevent user from getting stuck.
- Truncated long version label and added a tooltip for better user experience.

**Before**
Screenshot 2023-04-26 at 10 11 22 AM
Screenshot 2023-04-26 at 10 11 39 AM

**After**
Screenshot 2023-04-26 at 10 21 17 AM

Hover on tooltip to see full name
Screenshot 2023-04-26 at 10 21 28 AM

Loading View Diff, has back button to return to Version history 
Screenshot 2023-04-26 at 10 14 41 AM

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/71608/ui-feedback-from-installing-schrodinger-s-app

#### Does this PR introduce a user-facing change?

```release-note
UI Improvements made on version history page such as truncating long version label, removal of unnecessary preflight icon, and improve content layout.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: web/src/components/apps/AppVersionHistory.tsx [hunk 1]
```diff
@@ -99,7 +100,6 @@ type State = {
   errorMsg: string;
   errorTitle: string;
   firstSequence: Number | string;
-  hasPreflightChecks: boolean;
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:102]: replaced this with preflightState

File: web/src/components/apps/AppVersionHistory.tsx [hunk 2]
```diff
@@ -1687,6 +1698,40 @@ class AppVersionHistory extends Component<Props, State> {
       }
     }
 
+    const renderVersionLabel = () => {
+      let shorten = "";
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:1702]: truncating the version label

File: web/src/components/apps/AppVersionHistory.tsx [hunk 3]
```diff
@@ -1711,7 +1756,7 @@ class AppVersionHistory extends Component<Props, State> {
                   toggleErrorModal={resetRedeployErrorMessage}
                 />
               )}
-              {!gitopsIsConnected && (
+              {!gitopsIsConnected && !showDiffOverlay && (
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:1759]: hides this when showDiff is true

File: web/src/components/apps/AppVersionHistory.tsx [hunk 4]
```diff
@@ -1835,28 +1851,13 @@ class AppVersionHistory extends Component<Props, State> {
...
-                                    size={12}
-                                    className="version-row-preflight-status-icon warning-color"
-                                    style={{ left: "15px", top: "-6px" }}
-                                    color={""}
-                                    disableFill={false}
-                                    removeInlineStyle={false}
-                                  />
-                                </div>
-                              ) : null}
+                              <PreflightIcon
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:1854]: extracted PreflightIcon to it's own component

File: web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx [hunk 5]
```diff
@@ -270,21 +271,15 @@ function AppVersionHistoryRow(props: Props) {
     }
 
     const preflightState = getPreflightState(version);
-    let checksStatusText;
```
Comment (miaawong) [inline web/src/features/AppVersionHistory/AppVersionHistoryRow.tsx:273]: this logic is now inside of PreflightIcon component

File: web/src/features/VersionDiff/VersionDiff.tsx [hunk 6]
```diff
@@ -138,9 +138,31 @@ class VersionDiff extends React.Component<Props, State> {
 
     if (loadingFileTrees) {
       return (
-        <div className="u-height--full u-width--full flex alignItems--center justifyContent--center u-marginTop--15">
-          <Loader size="60" />
-        </div>
+        <>
```
Comment (miaawong) [inline web/src/features/VersionDiff/VersionDiff.tsx:141]: adds the back button when it's loading

**Review by rianfowler — APPROVED** [review]
nice work. extracting the icon logic is good cleanup

**Comment by rianfowler** [comment]
looks like this needs prettier ran on it

**Review by rianfowler — DISMISSED** [review]

**Review by rianfowler — APPROVED** [review]

**Pull request merged**